internal/host/validator.go

```diff
@@ -251,39 +253,46 @@ func (v *validator) printIsMachineCidrDefined(context *validationContext, status
 	}
 }
 
-func (v *validator) hasCpuCoresForRole(c *validationContext) ValidationStatus {
+func (v *validator) hasCPUCoresForRole(c *validationContext) ValidationStatus {
 	if c.inventory == nil {
 		return ValidationPending
 	}
-	switch c.host.Role {
-	case models.HostRoleMaster:
-		return boolValue(c.inventory.CPU.Count >= v.hwValidatorCfg.MinCPUCoresMaster)
-	case models.HostRoleWorker, models.HostRoleAutoAssign:
-		return boolValue(c.inventory.CPU.Count >= v.hwValidatorCfg.MinCPUCoresWorker)
-	default:
-		v.log.Errorf("Unexpected role %s", c.host.Role)
-		return ValidationError
+	if c.host.Role == models.HostRoleMaster || c.host.Role == models.HostRoleWorker || c.host.Role == models.HostRoleAutoAssign {
```

Comment:
Why not to check whether `c.host.Role != models.HostRoleBootstrap`?

Reply:
I was not successful at creating a test case with this role type, the logic didn't seem to support a state transition with bootstrap. It might be that there is a specific state in which it is supported, but after looking at the OCS validation code, I did not see any mention of the bootstrap as a type of role to check, so I dropped it altogether.